### PR TITLE
refactor: Fetch java executable location from PATH if JAVA_HOME is unset

### PR DESCRIPTION
**Pull request by mykola-mokhnach**
Related to https://github.com/appium/appium/issues/15693

### CODE REVIEW

**Review by KazuCocoa — APPROVED** [review]

File: lib/helpers.js [hunk 1]
```diff
@@ -166,8 +166,26 @@ const getJavaHome = _.memoize(async function getJavaHome () {
...
+  }
+  const executableName = `java${system.isWindows() ? '.exe' : ''}`;
+  if (javaHome) {
+    const resultPath = path.resolve(javaHome, 'bin', executableName);
+    if (await fs.exists(resultPath)) {
+      return resultPath;
+    }
+  }
+  try {
+    return await fs.which(executableName);
```
Comment (jlipps) [inline lib/helpers.js:184]: should we generalize this for other binaries, not just java?

**Review by jlipps — APPROVED** [review]

File: lib/helpers.js (hunk 1 above)
Comment (mykola-mokhnach) [reply lib/helpers.js:184]: which binaries exactly you'd like to generalize?

Comment (jlipps) [reply lib/helpers.js:184]: i'm not sure, don't we do similar logic currently for binaries inside ANDROID_HOME, and, e.g., jarsigner?

Comment (mykola-mokhnach) [reply lib/helpers.js:184]: jarsigner is only used as a fallback if apksigner fails for some reason, so I would rather leave it as is for now

**Pull request merged**